modules/luigi/__init__.py

```diff
@@ -161,16 +234,61 @@ def prompt_order(ctx: rs.ContextWrapper):
             return rs.Emit()
 
     @rs.state(
-        cond=sig_start_order_question.max_age(-1) & sig_yesno_detected,
+        cond=sig_suggested_ice_cream.max_age(-1) & sig_yesno_detected,
         read=nlp.prop_yesno,
-        write=rawio.prop_out)
+        write=rawio.prop_out,
+        signal=sig_start_order_question,
+        emit_detached=True)
     def analyse_ice_cream_suggestion_answer(ctx: rs.ContextWrapper):
         if ctx[nlp.prop_yesno] == "yes":
             ctx[rawio.prop_out] = verbaliser.get_random_successful_answer("greet_general")
-            # TODO list flavors that we have?
+            return rs.Emit()
         elif ctx[nlp.prop_yesno] == "no":
             ctx[rawio.prop_out] = verbaliser.get_random_failure_answer("greet_general")
 
+    @rs.state(
+        cond=sig_ice_cream_desire.max_age(-1),
+        write=rawio.prop_out,
+        signal=sig_start_order_question,
+        emit_detached=True)
+    def ice_cream_desire_will_be_fulfilled(ctx: rs.ContextWrapper):
+        ctx[rawio.prop_out] = "you are talking to the right person, i can get you some ice cream!"
+        return rs.Emit()
+
+    @rs.state(
+        cond=sig_start_order_question.max_age(-1) & sig_telegram_conversation.max_age(-1),
+        write=rawio.prop_out)
+    def ask_for_location(ctx: rs.ContextWrapper):
+        ctx[rawio.prop_out] = "just tell me where i can meet you!"
+
+    @rs.state(
+        cond=sig_start_order_question.max_age(-1) & sig_suggested_ice_cream.detached().max_age(-1)
+             | sig_has_arrived.detached().min_age(2),
+        write=rawio.prop_out)
+    def start_order(ctx: rs.ContextWrapper):
+        ctx[rawio.prop_out] = "what can i get you?"
+
+    @rs.state(
+        read=prop_location,
+        write=rawio.prop_out,
+        signal=sig_has_arrived, # TODO remove, this is just a dummy, should be emitted when AD team signals arrival
+        emit_detached=True)
+    def known_location(ctx: rs.ContextWrapper):
+        location = ctx[prop_location]
+        if location == "unknown":
+            ctx[rawio.prop_out] = "hmm, i didn't understand where you are... " \
+                                  "maybe i can meet you at a spot everyone knows?"
+        else:
+            ctx[rawio.prop_out] = "i will be at {location} in {min} minutes, see you then!" \
```

Comment:
Nice, I can add phrases to a yml file. Using verbaliser would be more flexible :+1:

Reply:
Yes, I normally just hard-code them first and then we can add more phrases later :)